src/code/reducers/index.js

```diff
@@ -73,6 +75,33 @@ export default function reducer(state, action) {
         challengeProgress: progress
       });
     }
+    case actionTypes.EGG_SELECTED: {
+      return state.merge({
+        selectedEggIndex: action.eggIndex
+      });
+    }
```

Comment:
Rather than having these selected indexes at the top level, why not add `selected` properties to members of the `drakes` array, and eventually the `basket` array (once that selection action is Redux-ified)? This would seem to be more generalizable to other activities where a user is selecting drakes, which will happen often, if we add the rest of the Geniverse activities. Selecting an egg could then call a totally-generic `SELECT_DRAKE` action.

Not only would this save us needing to put these indexes at the top level, it would be easier for your views. Instead of needing to pass around indexes, the object representing a `basket` simply has a property for whether it's selected, which seems more React-y.

(It adds a couple lines of code when you need to de-select all other items, but removes lines of code elsewhere.)

Of course, could be added to a template-specific section, but I don't think they need to be, since selecting drakes and other members of arrays is something we'll need to do again and again.